torch/csrc/jit/codegen/cuda/test/test_gpu_tensor_factories.cpp

```diff
@@ -0,0 +1,103 @@
+#if defined(USE_CUDA)
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <torch/csrc/jit/codegen/cuda/codegen.h>
+#include <torch/csrc/jit/codegen/cuda/executor.h>
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_iostream.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_cache.h>
+#include <torch/csrc/jit/codegen/cuda/ops/all_ops.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_gpu_validator.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_utils.h>
+
+// Tests go in torch::jit
+namespace torch {
+namespace jit {
+
+using namespace torch::jit::fuser::cuda;
+
+TEST_F(NVFuserTest, FusionStandaloneARange_CUDA) {
+  auto starts_ends = {-1., 0., 10.3, 1024. * 256};
+  auto steps = {-1.5, 1., 2.};
+  auto dtypes = {kFloat, kLong, kDouble};
+
+  for (auto dtype : dtypes) {
+    for (auto start : starts_ends) {
+      for (auto end : starts_ends) {
+        for (auto step : steps) {
+          const auto options =
+              at::TensorOptions().dtype(dtype).device(at::kCUDA, 0);
+
+          if (std::signbit(end - start) != std::signbit(step)) {
+            continue;
+          }
+
+          at::Tensor a =
+              at::arange((int64_t)start, (int64_t)end, (int64_t)step, options);
+          at::Tensor b =
+              at::arange((double)start, (double)end, (double)step, options);
+          at::Tensor c =
+              at::arange((int64_t)start, (double)end, (double)step, options);
+          at::Tensor d =
+              at::arange((double)start, (double)end, (int64_t)step, options);
+
+          auto fusion = std::make_unique<Fusion>();
+          FusionGuard fg(fusion.get());
```

Comment:
This fusion definition looks like it could get pulled out of the loops.

Reply:
done